beacon/engine/types.go

```diff
@@ -275,7 +275,7 @@ func ExecutableDataToBlockNoHash(data ExecutableData, versionedHashes []common.H
 	// ExecutableData before withdrawals are enabled by marshaling
 	// Withdrawals as the json null value.
 	var withdrawalsRoot *common.Hash
-	if data.WithdrawalsRoot != nil {
+	if config.IsOptimismIsthmus(data.Timestamp) && data.WithdrawalsRoot != nil {
 		if data.Withdrawals == nil || len(data.Withdrawals) != 0 {
 			return nil, fmt.Errorf("attribute WithdrawalsRoot was set. Expecting non-nil empty withdrawals list, but got %v", data.Withdrawals)
```

Comment:
There are a bunch of tests in `eth/catalyst/api_test.go` which explicitly set a withdrawals list assuming a post-shanghai block. Example: https://github.com/ethereum-optimism/op-geth/blob/optimism/eth/catalyst/api_test.go#L1114-L1131

These tests obviously fail because we require `data.Withdrawals` to be an empty list or nil. Passing in the chain config here allows us to check if we're at a Isthmus block and enforce the condition only if that's true.